### PR TITLE
config_psa: do not update legacy symbols in client-only PSA build

### PR DESCRIPTION
**Pull request by valeriosetti**
## Description

This commits belongs to PR 9138 of the official MbedTLS repo (https://github.com/Mbed-TLS/mbedtls/pull/9138). It will be included both in the next LTS release (3.6.1) as well as the next standard one (4.0) so it can be dropped at the next Zephyr's downstream version bump.

### CODE REVIEW

**Comment by valeriosetti** [comment]
Closing this PR because as shown in https://github.com/zephyrproject-rtos/zephyr/pull/76207 this PR seems to bring no benefit in reducing the NS image size.

**Comment by valeriosetti** [comment]
I'm closing this one because:

- commit https://github.com/zephyrproject-rtos/mbedtls/pull/58/commits/5a4107af060b06bec392509a147d6e07b3a49fb9 is already part of the official Mbed TLS 3.6.1 release which is part of https://github.com/zephyrproject-rtos/mbedtls/pull/60
- commit https://github.com/zephyrproject-rtos/mbedtls/pull/58/commits/ccac8b5fd4e7b8f1f388c366b8012870d7be1e70 was cherry-picked by https://github.com/zephyrproject-rtos/mbedtls/pull/60

So basically https://github.com/zephyrproject-rtos/mbedtls/pull/60 includes this PR, so this one can be closed.
